.azure/templates/build-test.yml

```diff
@@ -88,13 +88,13 @@ steps:
       set -e -x
       sudo apt-get install libacl1-dev libncurses5-dev pkg-config
       git clone --single-branch \
-                --branch "${ICEORYX_BRANCH:-master}" \
+                --branch "${ICEORYX_BRANCH:-v1.0.1}" \
```

Comment:
Pinning it to the `release_1.0` branch seems like a better compromise (it is what I use locally anyway). In my view it would be even better would be if we can handle both the 1.0.x and current master, and it seems like this could be done with a simple version check.

Reply:
@eboasson I think we can do a version check to determine if it is `1.0.x` or `master`, but if we choose the current master from `iceoryx` then probably a lot has been changed when compared with `1.0.x` and only fixing only this dependency might not be sufficient to make it work! 
Maybe we should sync the updating of `iceoryx` version here with `iceoryx` releases or we have a nightly job that tests with `iceoryx` master and then fix any test failures from this job.